depsync/syncer.go

```diff
@@ -0,0 +1,149 @@
+package depsync
+
+import (
+	"errors"
+	"fmt"
+	"github.com/go-kit/kit/log"
+	"github.com/micromdm/dep"
+	"time"
+)
+
+// Syncer is responsible for fetching devices from DEP, keeping them in sync, and storing cursor information about the
+// current fetch operation.
+type Syncer interface {
+	Start(deviceChan chan dep.Device)
+	Pause()
+	Resume()
+	Fetch(deviceChan chan dep.Device) (bool, error)
+}
+
+// Cursor represents information about the current fetch operation. It is required to fetch multiple "pages" of devices.
+type Cursor struct {
+	Value   string
+	Created *time.Time // >7 days cursor is invalid.
+}
+
+type syncer struct {
+	logger               log.Logger
+	client               dep.Client
+	tickerChan           <-chan time.Time
```

Comment:
lets add `fetchInterval time.Duration` and create an internal ticker instead. Both this and disabled seem unnecessary.